app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,175 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  const sjcl = require('sjcl');
+
+  function importFxaCryptoDeriver() {
+    return import(/* webpackChunkName: "fxaCryptoDeriver" */ 'fxaCryptoDeriver');
+  }
+
+  const RECOVERY_KEY_VERSION = '\x01';
+  const ALPHABET_BASE32 = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+
+  return {
+    /**
+     * Generate a random base32 string of length and
+     * prepend encryption version number.
+     *
+     * @param {Integer} length Length of string to generate
+     * @returns {string}
+     */
+    generateRecoveryKey: function (length) {
+      return importFxaCryptoDeriver().then((fxaCryptoDeriver) => {
```

Comment:
Since `jose` is the only thing used in the crypto deriver, in the function signature you can do:

```js
return importFxaCryptoDeriver().then(({ jose }) => {`
...
```